### PR TITLE
[Wallet] Payment limit bug fix

### PR DESCRIPTION
**Pull request by tarikbellamine**
### Description
Fixed bug that resulted in false errors regarding users being over the daily transaction limit. This was caused for two reasons:
- Amount was being incorrectly stored in redux as a non-primitive data type (BigNumber) yet the created interface was defining it as a number, which resulted in some faulty math
- Payments were being stored on transaction initiation, not success, which would mean failed transactions would incorrectly count against the daily limit

### Other changes
Minor cleanup

### Tested
Yes

### Related issues

- Part of #3988 

### Backwards compatibility
Created redux store migration.

### CODE REVIEW

File: packages/mobile/src/send/reducers.ts
Comment (cmcewen) [inline packages/mobile/src/send/reducers.ts:54]: Do you need to set `isSending` to false here?

File: packages/mobile/src/send/utils.ts
Comment (cmcewen) [inline packages/mobile/src/send/utils.ts]: you can define this outside the function as a constant

**Review by cmcewen — APPROVED** [review]
small comments but otherwise looks good!

**Comment by etuleu** [comment]
I don't have strong feelings on this one but I believe the initial reasoning for appending transactions on initiation instead of success was that an "abuser" could in theory initiate multiple transactions while on airplane mode that could add up to more than the limit. I don't know how big of a deal that is. Maybe we could just get @nityas attention on this just to be sure.

**Comment by i1skn** [comment]
> I don't have strong feelings on this one but I believe the initial reasoning for appending transactions on initiation instead of success was that an "abuser" could in theory initiate multiple transactions while on airplane mode that could add up to more than the limit. I don't know how big of a deal that is. Maybe we could just get @nityas attention on this just to be sure.

Considering that this limit is only imposed in the Valora app I would guess that real "abuser" would use other tools, cause this limit is not enforced on the blockchain level.

LGTM after adressing @cmcewen comments 👍 Great work on catching this!

**Review by i1skn — APPROVED** [review]

File: packages/mobile/src/send/reducers.ts
Comment (tarikbellamine) [reply packages/mobile/src/send/reducers.ts:54]: I would think so but that's not how it was setup before. Shouldn't hurt to add that in so I'll go ahead and do that.

File: packages/mobile/src/fiatExchanges/FiatExchangeAmount.test.tsx
Comment (jmrossy) [inline packages/mobile/src/fiatExchanges/FiatExchangeAmount.test.tsx]: Consider importing `ALERT_BANNER_DURATION` instead of hardcoding these

File: packages/mobile/src/send/utils.ts
Comment (jmrossy) [inline packages/mobile/src/send/utils.ts]: Dupe of https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/config.ts#L38

**Review by jmrossy — COMMENTED** [review]
Looks good, just two minor comments. Removing automerge so you will see them before this merges.

**Pull request merged**
